lib/src/protocol/pipe.rs

```diff
@@ -677,7 +677,7 @@ impl<Front: SocketHandler> Pipe<Front> {
         let tokens = self.tokens();
 
         if let Some(ref mut backend) = self.backend {
-            let (sz, r) = backend.socket_read(&mut self.back_buf.space());
+            let (sz, r) = backend.socket_read(self.back_buf.space());
```

Comment:
I you really want to please me, you can expand these obscure `sz` and `r` variable name. I guess `size` and… `remaining` ?